Commit directly after sharing a project in git fails

Steps to reproduce:
1. Create a new project
2. Use the Team -> Share project wizard from the Project Explorer view to share the project
3. Use the VirSat Navigator commit button to commit the files in the project

The following exception is printed:

`!ENTRY org.eclipse.e4.ui.workbench 4 0 2019-11-08 12:38:04.875
!MESSAGE Command 'de.dlr.sc.virsat.team.ui.command.commit' failed
!STACK 0
org.eclipse.core.commands.ExecutionException: Error executing 'de.dlr.sc.virsat.team.ui.command.commit': java.lang.IllegalArgumentException: Path must include project and resource name: /gitTest/
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:170)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:498)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:488)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4131)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1056)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3944)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3547)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:635)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:559)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:660)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:597)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1468)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1441)
Caused by: org.eclipse.e4.core.di.InjectionException: java.lang.IllegalArgumentException: Path must include project and resource name: /gitTest/
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:68)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:320)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:254)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
	... 33 more
Caused by: java.lang.IllegalArgumentException: Path must include project and resource name: /gitTest/
	at org.eclipse.core.runtime.Assert.isLegal(Assert.java:66)
	at org.eclipse.core.internal.resources.Workspace.newResource(Workspace.java:2123)
	at org.eclipse.core.internal.resources.Container.getFolder(Container.java:205)
	at de.dlr.sc.virsat.team.ui.action.git.GitCommitAction.createEmptyObjectsForEmptyFolders(GitCommitAction.java:126)
	at de.dlr.sc.virsat.team.ui.action.git.GitCommitAction.execute(GitCommitAction.java:80)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:283)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:95)
	at sun.reflect.GeneratedMethodAccessor48.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	... 37 more`

--- a/de.dlr.sc.virsat.team.ui/src/de/dlr/sc/virsat/team/ui/action/git/GitCommitAction.java
+++ b/de.dlr.sc.virsat.team.ui/src/de/dlr/sc/virsat/team/ui/action/git/GitCommitAction.java
@@ -123,22 +123,28 @@ public class GitCommitAction extends AbstractHandler {
 	 */
 	private void createEmptyObjectsForEmptyFolders(Set<String> untrackedFolders) throws CoreException {
 		for (String untrackedFolder : untrackedFolders) {
-			IFolder folder = ResourcesPlugin.getWorkspace().getRoot().getFolder(new Path(untrackedFolder));
-			if (folder.exists()) {
-				folder.accept(new IResourceVisitor() {
-					@Override
-					public boolean visit(IResource resource) throws CoreException {
-						if (resource instanceof IFolder) {
-							IFolder subFolder = (IFolder) resource;
-							boolean isEmptyFolder = subFolder.members().length == 0;
-							if (isEmptyFolder) {
-								IFile emptyFile = subFolder.getFile(EMPTY_FILE_NAME);
-								emptyFile.create(new ByteArrayInputStream(new byte[0]), IResource.NONE, null);
+			Path path = new Path(untrackedFolder);
+			// Workaround for the issue https://github.com/virtualsatellite/VirtualSatellite4-Core/issues/347
+			// A path with segment length 1 is considered to be a Project by eclipse.
+			// But the operation getFolder() is only applicable for Folders.
+			if (path.segmentCount() != 1) {
+				IFolder folder = ResourcesPlugin.getWorkspace().getRoot().getFolder(path);
+				if (folder.exists()) {
+					folder.accept(new IResourceVisitor() {
+						@Override
+						public boolean visit(IResource resource) throws CoreException {
+							if (resource instanceof IFolder) {
+								IFolder subFolder = (IFolder) resource;
+								boolean isEmptyFolder = subFolder.members().length == 0;
+								if (isEmptyFolder) {
+									IFile emptyFile = subFolder.getFile(EMPTY_FILE_NAME);
+									emptyFile.create(new ByteArrayInputStream(new byte[0]), IResource.NONE, null);
+								}
 							}
+							return true;
 						}
-						return true;
-					}
-				});
+					});
+				}
 			}
 		}
 	}

